src/components/buttons/secondary-button/secondary-button.js

```diff
@@ -38,65 +29,94 @@ export const getIconThemeColor = props => {
   return props.iconLeft.props.theme;
 };
 
-const getThemeClassName = theme => {
-  if (!theme) return undefined;
-
-  const themeClassName = styles[`theme-${theme}`];
-
-  // Whenever a theme is specified a fitting theme className should
-  // be available except for the default theme.
-  invariant(
-    themeClassName || theme === 'default',
-    `ui-kit/SecondaryButton: the specified theme '${theme}' is not supported.`
-  );
-
-  return themeClassName;
-};
-
 export const SecondaryButton = props => {
   const dataProps = {
     'data-track-component': 'SecondaryButton',
     ...filterDataAttributes(props),
   };
+  const isActive = props.isToggleButton && props.isToggled;
+  const shouldUseLinkTag = !props.isDisabled && Boolean(props.linkTo);
+
+  const containerStyles = [
+    css`
+      display: inline-block;
+      background-color: ${vars['--color-white']};
+      border-radius: ${vars['--border-radius-6']};
+      box-shadow: ${vars['--shadow-7']};
+      color: ${vars['--color-black']};
+      font-size: ${vars['--font-size-default']};
+      transition: background-color ${vars['--transition-linear-80ms']},
+        box-shadow ${vars['--transition-easeinout-150ms']};
+    `,
+    getStateStyles(props.isDisabled, isActive, props.theme),
+    getThemeStyles(props.theme),
+  ];
+
+  const containerElements = (
+    <AccessibleButton
+      type={props.type}
+      buttonAttributes={dataProps}
+      label={props.label}
+      onClick={props.onClick}
+      isToggleButton={props.isToggleButton}
+      isToggled={props.isToggled}
+      isDisabled={props.isDisabled}
+      css={css`
+        display: flex;
+        align-items: center;
+        padding: 0 ${vars['--spacing-16']};
+        height: ${vars['--big-button-height']};
+      `}
+    >
+      <Spacings.Inline alignItems="center" scale="xs">
+        {Boolean(props.iconLeft) && (
+          <span
+            css={css`
+              margin: 0 ${vars['--spacing-4']} 0 0;
+              display: flex;
+              align-items: center;
+              justify-content: center;
+            `}
+          >
+            {React.cloneElement(props.iconLeft, {
+              theme: getIconThemeColor(props),
+            })}
+          </span>
+        )}
+        <span>{props.label}</span>
+      </Spacings.Inline>
+    </AccessibleButton>
+  );
 
-  const shouldLink = !props.isDisabled && Boolean(props.linkTo);
-  const WrapperComponent = shouldLink ? Link : Div;
```

Comment:
I found it weird to have this component. Instead we should simply render the `<Link>` (with the `to` prop) or a normal `<div>`.